tests/openshell/conftest.py

```
@@ -0,0 +1,318 @@
+            "OPENAI_API_KEY": "unused",  # pragma: allowlist secret
+        }
+    }
+    if image:
+        LOGGER.info("Using custom sandbox image", image=image)
+        template_kwargs["image"] = image
+
+    spec = openshell_pb2.SandboxSpec(template=openshell_pb2.SandboxTemplate(**template_kwargs))
+    sandbox_name = generate_random_name(prefix="open-shell")
+
+    LOGGER.info("Creating OpenShell sandbox", name=sandbox_name)
+    sandbox_ref = sandbox_client.create(spec=spec, name=sandbox_name)  # noqa: FCN001
```

Ruff (0.16.1)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)